src/v/cluster/topics_frontend.cc

```diff
@@ -1083,16 +1191,14 @@ ss::future<std::error_code> topics_frontend::increase_replication_factor(
               co_return;
           }
 
-          auto partition_constraints = cluster::partition_constraints(
-            assignment.id, new_replication_factor);
+          partition_constraints.partition_id = assignment.id;
 
           auto ntp = model::ntp(topic.ns, topic.tp, assignment.id);
 
           auto reallocation = co_await _allocator.invoke_on(
             partition_allocator::shard,
-            [partition_constraints = std::move(partition_constraints),
-             assignment = std::move(assignment),
-             ntp](partition_allocator& al) {
+            [&partition_constraints, assignment = std::move(assignment), ntp](
```

Comment:
this will be a cross core access as it is capturing reference and accessing it cross cores, please capture a copy